packages/core-assortments/src/module/configureAssortmentsModule.ts

```diff
@@ -49,7 +49,7 @@ const buildFindSelector = ({
 
   if (tags) {
     if (Array.isArray(tags)) {
-      selector.tags = { $all: tags };
+      selector.tags = { $all: tags.map((tag) => new RegExp(`^${tag}$`, 'i')) };
```

Comment:
@Mikearaya Isn't it better to use

https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase

instead of a regex? like this it's using the regex search for every tags based lookup which might affect the performance.